### PR TITLE
Ticket 883/fix empty snb cache rewrite agent chached

### PR DESCRIPTION
**Pull request by sergiy-skalelabs**
Fixed problem related to empty SNB cache rewriting data.
No new tests needed.
No performance changes.

### CODE REVIEW

File: src/observer.mjs [hunk 1]
```diff
@@ -926,10 +913,8 @@ export async function cacheSChains( strChainNameConnectedTo, opts ) {
             opts.details.trace( "Will dispatch inThread-arrSChainsCached event in {}",
```
Comment (DmytroNazarenko) [inline src/observer.mjs]: Do we need `if( threadInfo.joCustomThreadProperties.isSChainsCacheNeeded )` here?

File: src/observer.mjs [hunk 2]
```diff
@@ -926,10 +913,8 @@ export async function cacheSChains( strChainNameConnectedTo, opts ) {
...
                 threadInfo.threadDescription() );
         }
-        events.dispatchEvent(
-            new UniversalDispatcherEvent(
-                "inThread-arrSChainsCached",
-                { "detail": { "arrSChainsCached": arrSChains } } ) );
+        events.dispatchEvent( new UniversalDispatcherEvent(
+            "inThread-arrSChainsCached", { "detail": { "arrSChainsCached": arrSChains } } ) );
         if( opts && opts.details ) {
```
Comment (DmytroNazarenko) [inline src/observer.mjs]: Do we need `if( threadInfo.joCustomThreadProperties.isSChainsCacheNeeded )` here?

**Review by DmytroNazarenko — CHANGES_REQUESTED** [review]

File: src/observer.mjs (hunk 2 above)
Comment (sergiy-skalelabs) [reply src/observer.mjs]: yes, we do not send un-needed notifications in wrong threads

File: src/observer.mjs (hunk 1 above)
Comment (sergiy-skalelabs) [reply src/observer.mjs]: yes, we do not send un-needed notifications in wrong threads

File: src/observer.mjs (hunk 2 above)
Comment (sergiy-skalelabs) [reply src/observer.mjs]: Fixed

**Review by DmytroNazarenko — APPROVED** [review]

**Pull request merged**